compiler/symbol.go

```diff
@@ -459,19 +462,35 @@ func isValidWasmType(typ types.Type, site wasmSite) bool {
 			return site == siteParam || site == siteIndirect
 		}
 	case *types.Array:
-		return site == siteIndirect && isValidWasmType(typ.Elem(), siteIndirect)
+		return site == siteIndirect && c.isValidWasmType(typ.Elem(), siteIndirect)
 	case *types.Struct:
 		if site != siteIndirect {
 			return false
 		}
+		// Structs with no fields do not need structs.HostLayout
+		if typ.NumFields() == 0 {
+			return true
+		}
+		hasHostLayout := true // default to true before detecting Go version
+		// (*types.Package).GoVersion added in go1.21
+		if gv, ok := any(c.pkg).(interface{ GoVersion() string }); ok {
+			if goenv.Compare(gv.GoVersion(), "go1.23") >= 0 {
+				hasHostLayout = false // package structs added in go1.23
```

Comment:
Interesting solution, I hadn't thought of that! Much cleaner than using build tags.

Reply:
Yeah! It's great when the only difference is a new method.